src/main/java/org/terasology/launcher/packages/PackageManager.java

```diff
@@ -94,44 +97,62 @@ public void sync() {
         localRepository.saveCache();
     }
 
-    // TODO: Move to constructor
-    public void initDatabase(Path launcherDir, Path gameDir) {
+    public void setupDirs(Path launcherDir, Path gameDir) {
         cacheDir = launcherDir.resolve(CACHE_DIRECTORY);
         installDir = gameDir.resolve(INSTALL_DIRECTORY);
-        final Path sourcesFile = launcherDir.resolve(SOURCES_FILENAME);
+        sourcesFile = launcherDir.resolve(SOURCES_FILENAME);
+    }
 
-        // Copy default sources file if necessary
+    public void validateSources() {
```

Comment:
Why are you using `void` here in combination with throwing an exception? I think it's easy to avoid throwing anything here by using a proper return type.

Reply:
I was refactoring the validation methods and just found out that there isn't really much need for throwing exceptions. We can simply use the logger to show which part of `sources.json` had a problem.